src/popup/components/Modals/ClaimGiftCard.vue

```diff
@@ -0,0 +1,406 @@
+    ).fee).shiftedBy(-AE_COIN_PRECISION);
+
+    const max = computed(() => balance.value.minus(fee));
+
+    async function updateBalance() {
+      const aeSdk = await getAeSdk();
+      balance.value = BigNumber((await aeSdk
+        .getBalance(address, { format: AE_AMOUNT_FORMATS.AE })
+        .catch(() => 0)
+      )
+        .toString());
+    }
+
+    const mainButtonText = computed(() => {
+      switch (step.value) {
+        case STEPS.initial:
+          return t('modals.claimGiftCard.redeemFull');
+        case STEPS.form:
+          return t('modals.claimGiftCard.redeem');
+        case STEPS.redeemPartially:
+        case STEPS.redeemFull:
+          return t('common.ok');
+        default:
+          return null;
+      }
+    });
+
+    function setMaxAmount() {
+      amount.value = max.value.isPositive() ? max.value.toString() : '0';
+    }
+
+    async function handleMainButtonClick() {
+      loading.value = true;
+      switch (step.value) {
+        case STEPS.initial:
+          setMaxAmount();
+          await store.dispatch('invites/claim', {
+            secretKey: props.secretKey,
+            recipientId: recipientId.value,
+            isMax: true,
+          });
+          step.value = STEPS.redeemFull;
+          break;
+        case STEPS.form:
+          await store.dispatch('invites/claim', {
+            secretKey: props.secretKey,
+            recipientId: recipientId.value,
+            amount: amount.value,
+            isMax: max.value.toString() === amount.value.toString(),
+          });
+          step.value = STEPS.redeemPartially;
+          break;
+        case STEPS.redeemPartially:
+        case STEPS.redeemFull:
+          props.resolve();
+          break;
+        default:
+          throw new Error(`Unknown step ${step.value}`);
+      }
+      loading.value = false;
+    }
+
+    const isDisabled = computed(() => {
+      switch (step.value) {
+        case STEPS.form:
+          return !amount.value || !!errors.value?.amount;
+        default:
+          return false;
+      }
+    });
+
+    const currencyFormatted = computed(() => getFormattedFiat(+amount.value));
+
+    onMounted(async () => {
+      await updateBalance();
+      if (balance.value.toNumber() === 0) {
+        step.value = STEPS.redeemFull;
+        isCardEmpty.value = true;
+      }
+      setInterval(updateBalance, 3000);
+    });
+
+    return {
+      AnimatedSpinner,
+      aeAccountsSelectOptions,
+      amount,
+      balance,
+      currencyFormatted,
+      errors,
+      fee,
+      handleMainButtonClick,
+      isCardEmpty,
+      isDisabled,
+      loading,
+      mainButtonText,
+      max,
+      PROTOCOL_AETERNITY,
+      recipientId,
+      setMaxAmount,
+      step,
+      STEPS,
+      updateBalance,
+    };
+  },
+});
+</script>
+
+<style lang="scss" scoped>
+@use '../../../styles/variables';
+@use '../../../styles/typography';
+
+.claim-gift-card {
+  :deep(.container) {
```

Comment:
Parent components should not affect child component internals. So please avoid using :deep selector. If you want to make a child component to behave or look differently please use bool prop and update the child component itself.